File: src/main/java/com/plantcare/plantcareassistant/entities/PlantWateringHistory.java
package com.plantcare.plantcareassistant.entities;

import java.time.LocalDateTime;
import jakarta.persistence.*;

import lombok.*;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "plant_watering_history")
public class PlantWateringHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_plant_id", nullable = false)
    private UserPlant userPlant;

    private LocalDateTime wateringDate;
    private Double waterAmount;
}

File: AndroidApp/app/src/main/java/com/example/plantcare/data/model/PlantWateringHistory.kt
package com.example.plantcare.data.model

import java.time.LocalDateTime

data class PlantWateringHistory(
    val id: Long,
    val userPlantId: Long,
    val wateringDate: LocalDateTime,
    val waterAmount: Double
)

data class WateringEventDto(
    val wateringDate: LocalDateTime,
    val waterAmount: Double
)

data class UserPlantWateringHistoryResponse(
    val userPlant: UserPlant,
    val wateringHistory: List<PlantWateringHistory>
)

